climpred/tests/test_stats.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+import numpy as np
+import xarray as xr
+
+from climpred.stats import xr_rm_trend
+
+
+def test_xr_rm_trend_2d_dataarray():
+    da = xr.DataArray(
+        np.vstack([np.arange(0, 5, 1),
+                   np.arange(0, 10, 2),
+                   np.arange(0, 40, 8),
+                   np.arange(0, 20, 4)]),
+        dims=['row', 'col']
+    )
+    da_dt = xr_rm_trend(da)
+    assert da_dt.shape == (4, 5)
+    # should have all values equal to 0 after detrending because it's linear
+    assert (xr_rm_trend(da, 'col') == 0).sum() == 20
+
+
+def test_xr_rm_trend_3d_dataset():
+    ds = xr.tutorial.open_dataset('air_temperature')

```

Comment:
I am afraid this test will take some time. Maybe just create sample files as in `test_prediction` or they way you did in the first test.